Unconditionally disable LTO in case it's injected from the outside.
LTO breaks the test as some symbols and sections are gone/rearranged.

https://github.com/boostorg/dll/issues/108

--- a/libs/dll/test/Jamfile.v2
+++ a/libs/dll/test/Jamfile.v2
@@ -36,7 +36,7 @@ project
       [ requires cxx11_static_assert ]
    # linux
       <target-os>linux:<linkflags>"-ldl"
-      <toolset>gcc:<cxxflags>"-Wall -Wextra -pedantic -Wno-long-long"
+      <toolset>gcc:<cxxflags>"-Wall -Wextra -pedantic -Wno-long-long -fno-lto"
 
    # others
       <local-visibility>hidden
